src/App.tsx

```diff
@@ -1,22 +1,69 @@
 /* eslint-disable jsx-a11y/control-has-associated-label */
-import React from 'react';
-import { UserWarning } from './UserWarning';
+import React, { useEffect, useState } from 'react';
+import classNames from 'classnames';
+import { Todo } from './types/Todo';
+import { TodoList } from './components/TodoList';
+import { FilterBy } from './enums/FilterBy';
+import { TodoFilter } from './components/TodoFilter';
+import { getTodos } from './api/todos';
+import { ErrorMessage } from './components/ErrorMessage/ErrorMessage';
 
-const USER_ID = 0;
+const USER_ID = 10321;
 
 export const App: React.FC = () => {
-  if (!USER_ID) {
-    return <UserWarning />;
-  }
+  const [todos, setTodos] = useState<Todo[]>([]);
+  const [filteredTodos, setFilteredTodos] = useState<Todo[]>([]);
+  const [filterBy, setFilterBy] = useState(FilterBy.All);
+  const [errorMessage, setErrorMessage] = useState('');
+
+  const activeTodosNumber = todos
+    .filter(todo => todo.completed === false).length;
+  const completedTodosNumber = todos.length - activeTodosNumber;
+
+  const getFilteredTodos = (filter: FilterBy) => {
+    switch (filter) {
+      case FilterBy.Active:
+        return todos.filter(todo => !todo.completed);
+
+      case FilterBy.Completed:
+        return todos.filter(todo => todo.completed);
+
+      default:
+        return todos;
+    }
+  };
+
+  const deleteErrorMessage = () => setErrorMessage('');
+
+  const loadData = async () => {
+    try {
+      const todosFromServer = await getTodos(USER_ID);
+
+      setTodos(todosFromServer);
+      setFilteredTodos(todosFromServer);
+    } catch {
+      setErrorMessage('Failed to load data');
+    }
+  };
+
+  useEffect(() => {
+    loadData();
+  }, []);
+
+  useEffect(() => setFilteredTodos(getFilteredTodos(filterBy)), [filterBy]);
```

Comment:
why do you need state for it?
You can just wrap you `getFilteredTodos` with `useMemo`